driver-core/src/main/java/com/datastax/driver/core/TupleType.java

```diff
@@ -100,6 +101,16 @@ public TupleValue newValue(Object... values) {
     }
 
     @Override
+    public boolean isFrozen() {
+        return true;
+    }
+
+    @Override
+    boolean canBeDeserializedAs(TypeToken typeToken) {
+        return typeToken.isAssignableFrom(getName().javaType);
+    }
+
+    @Override
     public int hashCode() {
         return Arrays.hashCode(new Object[]{ name, types });
     }
```

Comment:
In addition to these changes, the toString() method should be updated to surround the tuple with "frozen<>". This is required for Cassandra 2.1.0 to 2.1.2 and optional as of 2.1.3.